src/System.DirectoryServices/tests/System/DirectoryServices/ActiveDirectory/DomainControllerTests.cs

```diff
@@ -39,10 +39,10 @@ public void GetDomainController_InvalidName_ThrowsActiveDirectoryObjectNotFoundE
 
         [ConditionalFact(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsNanoServer))]
         [SkipOnTargetFramework(TargetFrameworkMonikers.Uap, "Access to path is denied when in App container")]
-        public void GetDomainController_InvalidIPV6_ThrowsInvalidCastException()
+        public void GetDomainController_InvalidIPV6_ThrowsActiveDirectoryObjectNotFoundException()
         {
             var context = new DirectoryContext(DirectoryContextType.DirectoryServer, "[::1]:port");
-            Assert.Throws<InvalidCastException>(() => DomainController.GetDomainController(context));
+            Assert.Throws<ActiveDirectoryObjectNotFoundException>(() => DomainController.GetDomainController(context));
```

Comment:
If there's the potential it could take that long, let's mark the test (and any others in a similar situation) as [Outerloop].